test.js

```diff
@@ -300,21 +300,20 @@ test('flush - send messages', async (t) => {
   const callbackB = spy();
   const callbackC = spy();
 
-  client.queue = [];
   client.identify({ userId: 'id', traits: { traitOne: 'a1' } }, callbackA);
   client.page({ userId: 'id', category: 'category', name: 'b1' }, callbackB);
   client.track({ userId: 'id', event: 'c1' }, callbackC);
 
   const data = await client.flush();
+  await delay(5); // ensure the test context exists long enough for the second flush to occur
```

Comment:
tested with a `delay(1)` and the behavior was still inconsistent, although more infrequent.